### ISSUE TITLE
Add optional exercises at the end of each section

### ISSUE THREAD

**Issue by svenvanderburg** [carpentries-incubator/python-intermediate-development#258]
### As a fast learner I want to have optional exercises/assignments at the end of each section.

When teaching this lesson in person (#159), we notice that some participants already finish a section when other participants are only halfway. We want everyone to start on a new section at the same time, so that we can also introduce some concepts for that section to everyone at the same time. It would be nice to have optional exercises/assignments at the end of each section for the fast learners.

**Comment by svenvanderburg** [comment]
We will likely try something out in our coming up workshop #259

**Comment by anenadic** [comment]
We welcome contributions to optional exercises.

**Comment by svenvanderburg** [comment]
@anenadic in the meantime we successfully trialed having optional exercises at the end of each chapter. If you want I can add them to the material in the summer (then I have loads of time).

**Comment by anenadic** [comment]
Yes, please! Much appreciated - will keep this issue open for a bit more then.

**Comment by svenvanderburg** [comment]
Cool, it's in my summer to-do list 🍹 🏖️
